### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by faddat**
changes only the linter, and **will** fail until gorockdb is updated

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -10,10 +10,9 @@ jobs:
     # We need to run the linter on the same image we use for building, since it
     # needs the C libraries installed for the dependencies to typecheck.
     runs-on: ubuntu-latest
-    container: tendermintdev/docker-tm-db-testing
+    container: archlinux
```
Comment (marbar3778) [inline .github/workflows/lint.yml:13]: why this change?.

Comment (faddat) [reply .github/workflows/lint.yml:13]: the former image only supports ancient rocksdb, and this prevents meaningful comparisons
